packages/trpc/server/routers/viewer/workflows.tsx

```diff
@@ -515,6 +520,9 @@ export const workflowsRouter = createProtectedRouter()
           });
           //step was edited
         } else if (JSON.stringify(oldStep) !== JSON.stringify(newStep)) {
+          if (user.plan === "FREE" && !isSMSAction(oldStep.action) && isSMSAction(newStep.action)) {
```

Comment:
So attackers can not add SMS actions to existing steps. It checks if the old action was already a SMS action and if not the new action also can not be a SMS action (for users on the FREE plan).